Memory tests are failing

```
$ go version
go version go1.10.1 linux/amd64
```

```
$ RUNTIME=kata-runtime FOCUS="hotplug memory when create containers" make -e docker
Running Suite: Integration Suite
================================
Random Seed: 1539975806
Will run 4 of 213 specs

SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
------------------------------
Hotplug memory when create containers Hotplug memory when create containers 
  hotplug memory when create containers should not fail
  /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46
Running command '/usr/bin/docker [docker ps -a -f name=BXmEBgyWnI6mmLaL299R2lc8PXBU2A --format {{.Status}}]'
[docker ps -a -f name=BXmEBgyWnI6mmLaL299R2lc8PXBU2A --format {{.Status}}]
Timeout: 60 seconds
Exit Code: 0
Stdout: 
Stderr: 
Could not read container ID file: open /tmp/cid072653181/BXmEBgyWnI6mmLaL299R2lc8PXBU2A: no such file or directory

• Failure in Spec Setup (BeforeEach) [0.048 seconds]
Hotplug memory when create containers
/home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:53
  Hotplug memory when create containers [BeforeEach]
  /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
    hotplug memory when create containers should not fail
    /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

    Expected
        <int64>: 0
    to be >
        <int>: 0

    /home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:63
------------------------------
Hotplug memory when create containers Hotplug memory when create containers 
  hotplug memory when create containers should not fail
  /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46
Running command '/usr/bin/docker [docker ps -a -f name=iYnRUaDI1DEtBQl6zrWZDFX6geI7DT --format {{.Status}}]'
[docker ps -a -f name=iYnRUaDI1DEtBQl6zrWZDFX6geI7DT --format {{.Status}}]
Timeout: 60 seconds
Exit Code: 0
Stdout: 
Stderr: 
Could not read container ID file: open /tmp/cid072653181/iYnRUaDI1DEtBQl6zrWZDFX6geI7DT: no such file or directory

• Failure in Spec Setup (BeforeEach) [0.048 seconds]
Hotplug memory when create containers
/home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:53
  Hotplug memory when create containers [BeforeEach]
  /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
    hotplug memory when create containers should not fail
    /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

    Expected
        <int64>: 0
    to be >
        <int>: 0

    /home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:63
------------------------------
Hotplug memory when create containers Hotplug memory when create containers 
  hotplug memory when create containers should not fail
  /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46
Running command '/usr/bin/docker [docker ps -a -f name=SASI43TJDqxn3MVKu82a6vOPSnC9qk --format {{.Status}}]'
[docker ps -a -f name=SASI43TJDqxn3MVKu82a6vOPSnC9qk --format {{.Status}}]
Timeout: 60 seconds
Exit Code: 0
Stdout: 
Stderr: 
Could not read container ID file: open /tmp/cid072653181/SASI43TJDqxn3MVKu82a6vOPSnC9qk: no such file or directory

• Failure in Spec Setup (BeforeEach) [0.054 seconds]
Hotplug memory when create containers
/home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:53
  Hotplug memory when create containers [BeforeEach]
  /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
    hotplug memory when create containers should not fail
    /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

    Expected
        <int64>: 0
    to be >
        <int>: 0

    /home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:63
------------------------------
Hotplug memory when create containers Hotplug memory when create containers 
  hotplug memory when create containers should not fail
  /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46
Running command '/usr/bin/docker [docker ps -a -f name=LsDA32avp7Zp0IndgTKFoEdtUN7Dih --format {{.Status}}]'
[docker ps -a -f name=LsDA32avp7Zp0IndgTKFoEdtUN7Dih --format {{.Status}}]
Timeout: 60 seconds
Exit Code: 0
Stdout: 
Stderr: 
Could not read container ID file: open /tmp/cid072653181/LsDA32avp7Zp0IndgTKFoEdtUN7Dih: no such file or directory

• Failure in Spec Setup (BeforeEach) [0.050 seconds]
Hotplug memory when create containers
/home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:53
  Hotplug memory when create containers [BeforeEach]
  /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table.go:92
    hotplug memory when create containers should not fail
    /home/x/go/src/github.com/kata-containers/tests/vendor/github.com/onsi/ginkgo/extensions/table/table_entry.go:46

    Expected
        <int64>: 0
    to be >
        <int>: 0

    /home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:63
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS

Summarizing 4 Failures:

[Fail] Hotplug memory when create containers [BeforeEach] Hotplug memory when create containers hotplug memory when create containers should not fail 
/home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:63

[Fail] Hotplug memory when create containers [BeforeEach] Hotplug memory when create containers hotplug memory when create containers should not fail 
/home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:63

[Fail] Hotplug memory when create containers [BeforeEach] Hotplug memory when create containers hotplug memory when create containers should not fail 
/home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:63

[Fail] Hotplug memory when create containers [BeforeEach] Hotplug memory when create containers hotplug memory when create containers should not fail 
/home/x/go/src/github.com/kata-containers/tests/integration/docker/mem_test.go:63

Ran 4 of 213 Specs in 0.202 seconds
FAIL! -- 0 Passed | 4 Failed | 0 Pending | 209 Skipped --- FAIL: TestIntegration (23.23s)
FAIL

Ginkgo ran 1 suite in 23.879706678s
Test Suite Failed



```

--- a/integration/docker/mem_test.go
+++ b/integration/docker/mem_test.go
@@ -6,14 +6,15 @@ package docker
 
 import (
 	"fmt"
-	. "github.com/kata-containers/tests"
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/ginkgo/extensions/table"
-	. "github.com/onsi/gomega"
 	"math"
 	"os"
 	"strconv"
 	"strings"
+
+	. "github.com/kata-containers/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
 )
 
 const (
@@ -26,16 +27,9 @@ const (
 	sysfsMemPath      = "/sys/devices/system/memory/"
 )
 
-func withDockerMemory(dockerMem int64, fail bool) TableEntry {
-	var msg string
-
-	if fail {
-		msg = "hotplug memory when create containers should fail"
-	} else {
-		msg = "hotplug memory when create containers should not fail"
-	}
-
-	return Entry(msg, dockerMem, fail)
+func withDockerMemory(dockerMem int64) TableEntry {
+	msg := "hotplug memory when create containers should not fail"
+	return Entry(msg, dockerMem)
 }
 
 func withUpdateMemoryConstraints(dockerMem int64, updateMem int64, fail bool) TableEntry {
@@ -55,12 +49,18 @@ var _ = Describe("Hotplug memory when create containers", func() {
 		args         []string
 		id           string
 		defaultMemSz int64
+		memBlockSize int64
+		exitCode     int
+		stdout       string
+		err          error
+		data         string
+		memBlockNum  int
 	)
 
 	BeforeEach(func() {
 		id = randomDockerName()
 		defaultMemSz = int64(KataConfig.Hypervisor[DefaultHypervisor].DefaultMemSz) << 20
-		Expect(defaultMemSz).To(BeNumerically(">", 0))
+		Expect(defaultMemSz).To(BeNumerically(">", int64(0)))
 	})
 
 	AfterEach(func() {
@@ -68,21 +68,21 @@ var _ = Describe("Hotplug memory when create containers", func() {
 	})
 
 	DescribeTable("Hotplug memory when create containers",
-		func(dockerMem int64, fail bool) {
+		func(dockerMem int64) {
 			args = []string{"--name", id, "-tid", "--rm", "-m", fmt.Sprintf("%d", dockerMem), Image}
-			_, _, exitCode := dockerRun(args...)
+			_, _, exitCode = dockerRun(args...)
 			Expect(exitCode).To(BeZero())
 
-			stdout, _, exitCode := dockerExec(id, "cat", memBlockSizePath)
+			stdout, _, exitCode = dockerExec(id, "cat", memBlockSizePath)
 			Expect(exitCode).To(BeZero())
-			data := strings.Trim(stdout, "\n\t ")
-			memBlockSize, err := strconv.ParseInt(data, 16, 64)
-			Expect(err).To(BeNil())
+			data = strings.Trim(stdout, "\n\t ")
+			memBlockSize, err = strconv.ParseInt(data, 16, 64)
+			Expect(err).ToNot(HaveOccurred())
 
 			stdout, _, exitCode = dockerExec(id, "sh", "-c", fmt.Sprintf("find %v -name memory* | wc -l", sysfsMemPath))
 			Expect(exitCode).To(BeZero())
-			memBlockNum, err := strconv.Atoi(strings.Trim(stdout, "\n\t "))
-			Expect(err).To(BeNil())
+			memBlockNum, err = strconv.Atoi(strings.Trim(stdout, "\n\t "))
+			Expect(err).ToNot(HaveOccurred())
 			memBlockNum--
 
 			mem := int64(math.Ceil(float64(dockerMem)/float64(memBlockSize))) * memBlockSize
@@ -90,10 +90,10 @@ var _ = Describe("Hotplug memory when create containers", func() {
 
 			Expect(RemoveDockerContainer(id)).To(BeTrue())
 		},
-		withDockerMemory(100*1024*1024, shouldNotFail),
-		withDockerMemory(200*1024*1024, shouldNotFail),
-		withDockerMemory(500*1024*1024, shouldNotFail),
-		withDockerMemory(1024*1024*1024, shouldNotFail),
+		withDockerMemory(100*1024*1024),
+		withDockerMemory(200*1024*1024),
+		withDockerMemory(500*1024*1024),
+		withDockerMemory(1024*1024*1024),
 	)
 })
 

